x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/job/categorization/GrokPatternCreatorTests.java

```diff
@@ -307,13 +307,44 @@ public void testFindBestGrokMatchFromExamplesGivenCatalinaLogs() {
                 + "Invalid chunk ignored."
         );
 
+        // For ECS compliant Grok patterns TOMCAT_DATESTAMP is defined as:
+        // TOMCAT_DATESTAMP (?:%{CATALINA8_DATESTAMP})|(?:%{CATALINA7_DATESTAMP})|(?:%{TOMCATLEGACY_DATESTAMP})
+        // and since the timestamps in the example messages are in CATALINA7_DATESTAMP format, TOMCAT_DATESTAMP, being at the
+        // front of our ORDERED_CANDIDATE_GROK_PATTERNS list, matches.
```

Comment:
I think a better way to fix this is to change `ORDERED_CANDIDATE_GROK_PATTERNS` to have `TOMCATLEGACY_DATESTAMP` first instead of `TOMCAT_DATESTAMP`.

Patterns that try multiple options are slower to match, and it seems like this old Tomcat format is really ancient as the person who updated the Grok patterns to ECS format couldn't find an example.

This discovery also has implications for the `TimestampFormatFinder` class.  That should also be changed to swap out `TOMCAT_DATESTAMP` for `TOMCATLEGACY_DATESTAMP` when ECS compatibility is set to `v1` - please open a separate PR for that.